File: Arrays/secondLargest.java


public class secondLargest {
    public static void main(String[] args) {
        int[] arr={77,52,85,745,965,1,235};
        int max=arr[0];
        int secondMax=Integer.MIN_VALUE;
        for(int i=0;i<arr.length;i++){
            if(arr[i]>max){
                secondMax=max;
                max=arr[i];
            }
            else if(arr[i]<max && arr[i]>secondMax){
                secondMax=arr[i];
            }
        }
        System.out.println("Largest value is:"+max);
        System.out.println("second largest is:" +secondMax);
    }
    
}

File: Arrays/secondLargest.py
def secondLargest(arr):
  large=arr[0]
  slarge=0
  for i in arr:
    if i>large:
      slarge=large
      large=i
    elif i<large and i>slarge:
      slarge=i
  return slarge

arr=[10,5,8,20]
print(secondLargest(arr))